tests/test.ts

```diff
@@ -447,3 +448,112 @@ describe('Test clasp logout function', () => {
+      CLASP, ['-h'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain(CLASP_USAGE);
+    expect(result.status).to.equal(0);
+  });
+});
+describe('Test clasp version function', () => {
+  it('should show version correctly', () => {
+    const result = spawnSync(
+      CLASP, ['--version'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain(require('./../package.json').version);
+    expect(result.status).to.equal(0);
+  });
+  it('should show help correctly', () => {
+    const result = spawnSync(
+      CLASP, ['-v'], { encoding: 'utf8' },
+    );
+    expect(result.stdout).to.contain(require('./../package.json').version);
+    expect(result.status).to.equal(0);
+  });
+});
+describe('Test clasp unknown function', () => {
+  it('should show version correctly', () => {
+    const result = spawnSync(
+      CLASP, ['unknown'], { encoding: 'utf8' },
+    );
+    expect(result.stderr).to.contain('🤔  Unknown command "unknown"');
+    expect(result.status).to.equal(1);
+  });
+});
+
+describe('Test all functions while logged out', () => {
+  const expectNoCredentials = (command: string) => {
+    const result = spawnSync(
+      CLASP, [command], { encoding : 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    expect(result.stderr).to.include(ERROR.NO_CREDENTIALS);
+  };
+  it('should fail to list (no credentials)', () => expectNoCredentials('list'));
+  it('should fail to clone (no credentials)', () => expectNoCredentials('clone'));
+  it('should fail to push (no credentials)', () => expectNoCredentials('push'));
+  it('should fail to deployments (no credentials)', () => expectNoCredentials('deployments'));
+  it('should fail to deploy (no credentials)', () => expectNoCredentials('deploy'));
+  it('should fail to version (no credentials)', () => expectNoCredentials('version'));
+  it('should fail to versions (no credentials)', () => expectNoCredentials('versions'));
+  // TODO: all test should have same order of checks
+  // and should all return ERROR.NO_CREDENTIALS
+  it('should pull error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['pull'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    // Should be ERROR.NO_CREDENTIALS
+    // see: https://github.com/google/clasp/issues/278
+    expect(result.stderr).to.contain(ERROR.SCRIPT_ID_DNE);
+  });
+  it('should open error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['open'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    // Should be ERROR.NO_CREDENTIALS
+    // see: https://github.com/google/clasp/issues/278
+    expect(result.stderr).to.contain(ERROR.SCRIPT_ID_DNE);
+  });
+  it('should redeploy error correctly', () => {
+    const result = spawnSync(
+      CLASP, ['redeploy', '1234'], { encoding: 'utf8' },
+    );
+    expect(result.status).to.equal(1);
+    expect(result.stderr).to.contain('error: missing required argument `version\'');
```

Comment:
Why does this have a \` character here?
Can we make this contain check have a more readable expected value?

Reply:
This is a feature of CommandJS, see https://github.com/tj/commander.js/issues/840

I've updated to skip these tests with a note to see that issue.